pkg/geoipupdate/database/http_reader.go

```diff
@@ -48,10 +48,9 @@ func NewHTTPReader(
 	retryFor time.Duration,
 	verbose bool,
 ) Reader {
-	transport := http.DefaultTransport
+	transport := &http.Transport{Proxy: http.ProxyFromEnvironment}
```

Comment:
This is missing all the other settings in `http.DefaultTransport`. It might be best to just clone the default transport instead.